tests/main/devtools-spec.ts

```diff
@@ -1,47 +1,59 @@
 /**
- * @jest-environment node
+ * @vitest-environment node
  */
-
-import { mocked } from 'jest-mock';
+import { type Mock, beforeAll, describe, expect, it, vi } from 'vitest';
 
 import { setupDevTools } from '../../src/main/devtools';
 import { isDevMode } from '../../src/main/utils/devmode';
 
-jest.mock('../../src/main/utils/devmode');
+vi.mock('../../src/main/utils/devmode');
+
+const installExtensionPromise = vi.hoisted(async () => {
+  const { mockRequire } = await import('../utils.js');
+  const installExtension = vi.fn();
 
-jest.mock('electron-devtools-installer', () => ({
-  default: jest.fn(),
-  REACT_DEVELOPER_TOOLS: 'REACT_DEVELOPER_TOOLS',
-  REACT_PERF: 'REACT_PERF',
-}));
+  mockRequire('electron-devtools-installer', {
+    default: installExtension,
+    REACT_DEVELOPER_TOOLS: 'REACT_DEVELOPER_TOOLS',
+    REACT_PERF: 'REACT_PERF',
+  });
+
+  return installExtension;
+});
 
 describe('devtools', () => {
-  it('does not set up developer tools if not in dev mode', () => {
-    const devtools = require('electron-devtools-installer');
-    mocked(isDevMode).mockReturnValue(false);
-    setupDevTools();
+  let installExtension: Mock;
 
-    expect(devtools.default).toHaveBeenCalledTimes(0);
+  beforeAll(async () => {
+    installExtension = await installExtensionPromise;
   });
 
-  it('sets up developer tools if in dev mode', () => {
-    const devtools = require('electron-devtools-installer');
-    mocked(isDevMode).mockReturnValue(true);
-    setupDevTools();
+  it('does not set up developer tools if not in dev mode', async () => {
+    vi.mocked(isDevMode).mockReturnValue(false);
+    await setupDevTools();
 
-    expect(devtools.default).toHaveBeenCalledTimes(1);
+    expect(installExtension).toHaveBeenCalledTimes(0);
+  });
+
+  it('sets up developer tools if in dev mode', async () => {
+    vi.mocked(isDevMode).mockReturnValue(true);
+    await setupDevTools();
+
+    expect(installExtension).toHaveBeenCalledTimes(1);
   });
 
   it('catch error in setting up developer tools', async () => {
-    const devtools = require('electron-devtools-installer');
+    console.warn = vi.fn();
+    const error = new Error('devtool error');
+
     // throw devtool error
-    devtools.default.mockRejectedValue(new Error('devtool error'));
-    mocked(isDevMode).mockReturnValue(true);
-
-    try {
-      await setupDevTools();
-    } catch (e) {
-      expect(e).toMatch('error');
-    }
+    vi.mocked(installExtension).mockRejectedValue(error);
+    vi.mocked(isDevMode).mockReturnValue(true);
+
+    await setupDevTools();
+    expect(console.warn).toHaveBeenCalledWith(
+      expect.stringMatching(/installDevTools/),
+      error,
+    );
```

Comment:
This test wasn't actually testing what it intended to, since the `expect` inside of the `catch` block never actually ran. Inside of `setupDevTools` any exceptions are caught and logged with `console.warn`, so this test is updated to actually test that.